### ISSUE TITLE
[8.0.0-beta2-rev7] When we select cell 0,0 and hide the data clicking on column A with RMB allows us to add new rows

### ISSUE THREAD

**Issue by AMBudnik** [handsontable/handsontable#7051]
### Description

As mentioned in this issue https://github.com/handsontable/handsontable/issues/7050 there is a difference between the menu list of options when cell `0,0` is being selected.

### Steps to reproduce

1. Click with <kbd>RMB</kbd> on the header `A` to open the context menu

Result 8.0.0-beta2-rev16 
![Zrzut ekranu 2020-06-25 o 14 30 29](https://user-images.githubusercontent.com/10757813/85720216-851e8e80-b6f0-11ea-84f1-1b5ad99baa7d.png)
https://jsfiddle.net/ph4b78uk/2/ 8.0.0-beta2-rev16
https://jsfiddle.net/ph4b78uk/4/ 8.0.0-beta2-rev 7

Result 7.4.2
![742-00-none](https://user-images.githubusercontent.com/10757813/85720238-894aac00-b6f0-11ea-8ca1-e0fde423afc3.png)
https://jsfiddle.net/g4xyLj32/1/ 7.4.2
https://jsfiddle.net/ph4b78uk/3/ 8.0.0-beta2-rev6

### Your environment
* Handsontable version: 8.0.0-beta2-rev7
* Browser Name and version: Chrome 83
* Operating System: macOS Catalina

**Comment by aninde** [comment]
This issue is still broken on merge-candidate for `8.0.0-beta2``PR#7027` https://jsfiddle.net/aninde/u1zmbfkd/

**Comment by wszymanski** [comment]
Another example, a simpler one:

```js
data: Handsontable.helper.createSpreadsheetData(5, 5),
rowHeaders: true,
colHeaders: true,
hiddenColumns: {
  columns: [1],
  indicators: true,
},
```

```js
hot.selectCell(2, 1)
```

Demos did by @aninde:

https://jsfiddle.net/aninde/y64gd5zk `7.4.2`
https://jsfiddle.net/aninde/gf2Lctdm `8-beta2-rev16`

**Comment by budnix** [comment]
This PR https://github.com/handsontable/handsontable/pull/7053 fixes this issue.

**Comment by AMBudnik** [comment]
**The main issue** does not differ for version 8.0.0-beta2-rev18 after merging https://github.com/handsontable/handsontable/pull/7053 https://jsfiddle.net/Lrtehqv9/ 


**@wszymanski 's case solved** in 8.0.0-beta2-rev18 https://jsfiddle.net/e2y0fx98/

**Comment by wszymanski** [comment]
I've added my example because it is the main reason why there is no header selection. On Handsontable 7 after a click on the border, there was a selection on the first element, **even it was hidden**. Now, click on the header select all cells and all column headers should be highlighted. We talked about it [here](https://github.com/handsontable/handsontable/pull/7027#issuecomment-651640272) and @budnix said that he will create a [separate issue for it](https://github.com/handsontable/handsontable/pull/7027#issuecomment-651642633).

FYI: @wojciechczerniak.

**Comment by wszymanski** [comment]
I think that the current behavior of the table is regression. The `selectAll` method selected all headers in such a case on the `master`.

**Comment by wszymanski** [comment]
Related to: https://github.com/handsontable/handsontable/issues/7063.

**Comment by AMBudnik** [comment]
I'm sharing additional details as this issue might have caused some misunderstanding.

First of all, as @budnix solved the selection of a header we will compare version `8.0.0-beta2-rev18` https://jsfiddle.net/emja9h6s/ with `7.4.2` https://jsfiddle.net/dnypv40j/

### Settings
In this case, we load 4 rows of data and hide them all programmatically via `hiddenRows` settings.
At the same time, we add `selectCell()` method on cell `0, 0` (which is hidden). 

### User action
The only use action is to open a context menu using <kbd>RMB</kbd> on header `A`. And here we can see that before `rev7` of `8.0.0-beta2` when we did that actions

* `Insert row above`
* `Insert row below`

are disabled, in `8.0.0-beta2-rev7` and newer those options are active.

### Screenshots

Here's `8.0.0-beta2-rev18`
![Zrzut ekranu 2020-07-1 o 14 44 23](https://user-images.githubusercontent.com/10757813/86245200-6c5b2080-bba9-11ea-957c-85be5560a869.png)

Here's `7.4.2`
![Zrzut ekranu 2020-07-1 o 14 43 45](https://user-images.githubusercontent.com/10757813/86245208-711fd480-bba9-11ea-9145-d62996478a6f.png)

### Debugging
#### `afterOnCellMouseDown` logs. In both cases when we click the header we get 
* 2nd parameter `{row: -1, col: 0}`
* 3rd paramteter `<th class=​"ht__highlight">​…​</th>​`

Let me know if you need a further investigation on this case.

**Comment by aninde** [comment]
**Case 2**

We can not add anything from the corner (top-left element)

https://jsfiddle.net/aninde/48z0jkt3/ `8.0.0-beta2-rev19`
https://jsfiddle.net/aninde/o5y19s6m/ `v7.4.`
![Screenshot 2020-07-07 at 13 35 06](https://user-images.githubusercontent.com/30186818/86776607-a1a7b880-c058-11ea-8f11-5d5f92aed154.png)
![Screenshot 2020-07-07 at 13 34 54](https://user-images.githubusercontent.com/30186818/86776614-a2d8e580-c058-11ea-9d90-214ffd29fc6a.png)

**Comment by aninde** [comment]
### Case 1
**solved**  https://github.com/handsontable/handsontable/issues/7051#issue-645526703

The original demo had incorrect classes. https://jsfiddle.net/aninde/c5m21px0/ - broken `8.0.0-rev16` 

This case is solved by `8.0.0-beta2-rev22` https://jsfiddle.net/aninde/09kg8o6q/.
The header is selected. We can't add any row from the header. Lack of bottom border is reported as a
regression.

![Screenshot 2020-07-10 at 16 22 32](https://user-images.githubusercontent.com/30186818/87165779-e63a8a80-c2ca-11ea-8fd3-7b1c7ae668dc.png)

### Case 2 
 **still works well** https://github.com/handsontable/handsontable/issues/7051#issuecomment-649521679
@wszymanski case is still fixed like it was with `rev18` https://jsfiddle.net/aninde/Lasu23rg/ in `rev22`

### Case 3 
**still works well** https://github.com/handsontable/handsontable/issues/7051#issuecomment-652405131

Still works well with `rev22` https://jsfiddle.net/aninde/g7bpy2uj/ 
 **afterOnCellMouseDown** logs look similar after clicking A header.
- 2nd parameter {row: -1, col: 0}
- 3rd paramteter <th class=​"ht__highlight ht__active_highlight">​…​</th>​

### Case 4 
I need to check assumptions 
https://github.com/handsontable/handsontable/issues/7051#issuecomment-654801324
https://jsfiddle.net/aninde/yz2r49en/ - `rev22`
![Screenshot 2020-07-10 at 16 40 24](https://user-images.githubusercontent.com/30186818/87166978-9ceb3a80-c2cc-11ea-8427-edc1fcbccee8.png)

**Comment by aninde** [comment]
Ok, Case 4 if correct with assumptions described https://github.com/handsontable/handsontable/pull/7082#issue-445323799  so I'm closing this regression.
<img width="440" alt="Screenshot 2020-07-10 at 16 53 08" src="https://user-images.githubusercontent.com/30186818/87167816-da9c9300-c2cd-11ea-9f10-5ed234a728d3.png">
